homeassistant/components/automation/__init__.py

```diff
@@ -442,6 +442,7 @@ async def async_trigger(self, run_variables, context=None, skip_condition=False)
         This method is a coroutine.
         """
         reason = ""
+        run_variables["automation_entity_id"] = self.entity_id
```

Comment:
Can you do something like this instead:
```suggestion
        variables["self"] = self.hass.states.get(self.entity_id).as_dict()
```

Then add the same to script, in `homeassistant.components.script.async_turn_on()`